becquerel/tools/xcom.py

```diff
@@ -300,10 +300,17 @@ def update(self, **kwargs):
             _XCOMQuery._check_z(znum)
             self._data['ZNum'] = '{:d}'.format(int(znum))
         elif 'compound' in kwargs:
-            self._method = '2'
+            # convert compound to mixture to avoid occasional problem
+            # with XCOM compound queries (see issue #76)
+            # self._method = '2'
+            self._method = '3'
             formula = kwargs['compound']
             _XCOMQuery._check_compound(formula)
-            self._data['Formula'] = formula
+            formulae = [formula + ' 1']
```

Comment:
Actually, `_check_compound` only makes sure the formula is alphanumeric, not that it is a string. I will add a check for the type.

Reply:
I take it back, the method `_argument_type` checks for a string and that is how `_XCOMQuery` determines whether the argument is a compound, before `_check_compound` is ever called:

https://github.com/lbl-anp/becquerel/blob/d601b916ff4ad8da7a94954f2366e09ca6979505/becquerel/tools/xcom.py#L214-L215

Reply:
Thanks for pointing that out! With that said, couldn't this be simplified:

```python
    def _argument_type(arg):
        """Determine if argument is a symbol, Z, compound, or mixture."""
        if isinstance(arg, string_types):
            if arg.isdigit():
                return {'z': arg}
            elif arg.lower() in [s.lower() for s in element.SYMBOLS]:
                return {'symbol': sym}
            else:
                return {'compound': arg}
        elif isinstance(arg, Iterable):
            return {'mixture': arg}
        raise XCOMInputError(
            'Cannot determine if argument {}'.format(arg) +
            ' is a symbol, Z, compound, or mixture')
```

This could be moved to future work...

Reply:
Yes, that would simplify the symbol checking but miss the case where Z is an int.

Reply:
Actually there are a couple more lines of code if I add the case where Z is an int...

Reply:
How about:

```python
    def _argument_type(arg):
        """Determine if argument is a symbol, Z, compound, or mixture."""
        if isinstance(arg, string_types):
            if arg.isdigit():
                return {'z': arg}
            elif arg.lower() in [s.lower() for s in element.SYMBOLS]:
                return {'symbol': sym}
            else:
                return {'compound': arg}
        elif isinstance(arg, int):
            return {'z': arg}
        elif isinstance(arg, Iterable):
            return {'mixture': arg}
        raise XCOMInputError(
            'Cannot determine if argument {}'.format(arg) +
            ' is a symbol, Z, compound, or mixture')
```